corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -600,13 +600,23 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
             $.when(fetchingPrompts).done(function (response) {
                 // Update models based on response
-                _.each(response.models, function (responseModel, i) {
-                    self.collection.models[i].set({
-                        error: responseModel.get('error'),
-                        required: responseModel.get('required'),
-                        required_msg: responseModel.get('required_msg'),
-                    });
+                if (response.queryResponse != null) {
+                    _.each(response.queryResponse.displays, function (responseModel, i) {
+                        self.collection.models[i].set({
+                            error: responseModel.error,
+                            required: responseModel.required,
+                            required_msg: responseModel.required_msg,
+                        });
                 });
+                } else {
+                    _.each(response.models, function (responseModel, i) {
+                        self.collection.models[i].set({
+                            error: responseModel.get('error'),
+                            required: responseModel.get('required'),
+                            required_msg: responseModel.get('required_msg'),
+                        });
+                    });
+                }
```

Comment:
nit: is there a way for this to be more concise? I see it's not exactly the same actions in both conditions, but in my head it's something like
```javascript
                let responseModels = response.models;
                if (response.queryResponse != null) {
                    responseModels = response.queryResponse.displays;
                }
                _.each(responseModels, function (responseModel, i) {
                ...

Reply:
ha yes I tried this, unfortunately they are totally different formats and I played around with getting the formatting right but settled on this given the urgency of this ticket, but I'm still tinkering with it